app/src/ui/lib/configure-git-user.tsx

```diff
@@ -42,27 +45,64 @@ export class ConfigureGitUser extends React.Component<
   public constructor(props: IConfigureGitUserProps) {
     super(props)
 
-    this.state = { name: '', email: '', avatarURL: null }
+    this.state = {
+      globalUserName: null,
+      globalUserEmail: null,
+      name: '',
+      email: '',
+      avatarURL: null,
+    }
   }
 
   public async componentWillMount() {
```

Comment:
could the contents of this method go in the constructor as part of initializing state? `componentWillMount` is a deprecated react lifecycle method so it would be nice to remove it.

the `prevState` in `setState` here should just be whatever was set in the constructor.

Reply:
Strictly speaking, no, they can't go into the constructor because of the async nature of the method. We can't use `setState` until the component is mounted and unless we implement another anti-pattern (isMounted) we won't know that. It would also be unsafe to update `this.state` again for the same reason. So it's a chicken and egg problem. Technically the same can be said about what we've got now though because we don't unsubscribe from the promises when the component gets unmounted but that's a problem we've got all over the code base.

Your point about `componentWillMount` is well taken though and I can update it to use `componentDidMount` instead.